tool/release/createnotes/Version.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2021 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.vaticle.dependencies.tool.release.createnotes
+
+data class Version(val major: Int, val minor: Int, val patch: Int, val alpha: Int?): Comparable<Version> {
+    companion object {
+        fun parse(version: String): Version {
+            val version1 = version.split("-")
+            require(version1.isNotEmpty() && version1.size <= 3) {
+                "version '$version1' does not follow the form 'x.y.z', 'x.y.z-alpha', or 'x.y.z-alpha-n'"
+            }
+            val version2 = version1[0].split(".")
+            if (version1.size == 1) {
+                require(version2.size == 3) { "version must be of the form x.y.z" }
+                return Version(
+                    major = version2[0].toInt(),
+                    minor = version2[1].toInt(),
+                    patch = version2[2].toInt(),
+                    alpha = null
+                )
+            } else if (version1.size == 2) {
```

Comment:
This `else if` branch handles when we get a string such as `2.0.0-alpha`. It implies that this is the first alpha and hence set `alpha = 1` when instantiating the `Version` class.